README.md

```diff
@@ -43,6 +43,14 @@ if (!map.has(key)) {
 map.get(key).doThing();
 ```
 
+With this proposal:
+
+```js
+map.upsert(key, o => o, () => value).doThing();
```

Comment:
Though `o=>o` could work, i have to say it's a little bit weird.

And i was caught by `() => {}`, I suspect it could be easily to find in code review when u see 

```js
const noop = new Function()

...
map.upsert(key, noop, () => value)
```

Reply:
The problem with noop and not identity is updating of immutable/primitive values:

```mjs
map = new Map();
map.upsert('foo', old => old + 1, () => 0);
```